taichi/ir/statements.h

```diff
@@ -303,6 +303,30 @@ class GlobalPtrStmt : public Stmt {
   TI_DEFINE_ACCEPT_AND_CLONE
 };
 
+/**
+ * An accessing tensor element operation.
```

Comment:
I don't quite understand what is a "tensor" in Taichi... Is it a Taichi field? What's the difference between this one and a GlobalPtrStmt? What do the fields `origin` and `offset` mean?

Reply:
The goal is to support dynamic indexing. We are now trying to access all elements by offset from the first element's address and the target element's index.

Reply:
Although probably intuitive in the SSA form IR, I think it's better to stress here that this statement doesn't modify `origin` -- it's just `origin + offset`, not `origin += offset`.

Reply:
Maybe `PtrOffsetStmt`? Also I'd suggest to drop `Global`. IIUC, this should also work for `alloca` later.

Reply:
https://github.com/taichi-dev/taichi/pull/2543/commits/3c20d4828a45cdb6e1396564fb49835e69eb86ab